Source/pyboy/botsupport/tilemap.py

```
@@ -0,0 +1,193 @@
+
+        Use this instead of `pyboy.botsupport.tilemap.TileMap.get_tile_index`, if you want a simple way to identify tiles without researching the quirks of tile indexes used in the Game Boy.
+
+        Returns:
+            int: Tile identifier.
+        """
+
+        tile = self.mb.getitem(self.get_tile_address(x,y))
+        if self.signed_tile_data:
+            return ((tile ^ 0x80) - 128) + 0xFF
+        else:
+            return tile
+
+    def get_tile_index(self, x, y):
+        """
+        Returns the index of the tile at the given coordinate in the tile map.
+
+        Consider using `pyboy.botsupport.tilemap.TileMap.get_tile_identifier` if you want a simple way to identify tiles in the tile map.
+
+        You can read how the indexes work in the [Pan Docs: VRAM Tile Data](http://bgb.bircd.org/pandocs.htm#vramtiledata).
+
+        Returns:
+            tuple (bool, int): First value indicates if the index is signed. The second value is the index. Both are needed for comparison.
+        """
+
+        tile = self.mb.getitem(self.get_tile_address(x,y))
+        if self.signed_tile_data:
+            return (self.signed_tile_data, (tile ^ 0x80) - 128)
+        else:
+            return (self.signed_tile_data, tile)
+
+    def get_tile_matrix(self):
+        """
+        Returns a matrix of 32x32 of the given tile map. Each element in the matrix, is the tile index to be shown on screen for each position.
+
+        The index can be used to quickly identify what is on the screen through this tile view.
+
+        Use `pyboy.botsupport.tilemap.TileMap.signed_tile_index` to test if the indexes are signed for this tile view. You can read how the indexes work in the [Pan Docs: VRAM Tile Data](http://bgb.bircd.org/pandocs.htm#vramtiledata).
+
+        Returns:
+            list: Nested list creating a 32x32 matrix of tile indexes.
+        """
+        return self[:,:]
+
+    def __str__(self):
+        adjust = 4
+        return (
```

```suggestion
        try:
        self.use_tile_objects(True)
            return (
```

---

Why is `try` needed? How would it fail?

---

I can see the point. I almost went with it, but I thought it was more clear without the `try`. Besides, any error would be silenced, which is not really that good.